### PR TITLE
Better findRoad JS

### PR DESCRIPTION
**Pull request by sonnyp**
This is not much better but reduce indentation

thanks for http://assil.me/2015/11/07/roadomatic-node-vs-go.html

### CODE REVIEW

File: server/backend/server.js [hunk 1]
```diff
@@ -76,33 +75,32 @@ var findRoad = (parsed, db, callback) => {
...
-
-        callback(resp);
-      });
+    if (!segment) {
+      if (err) {
+        // Collection read error
+        console.error(err);
+        resp.online = 0;
+      }
+      return callback(resp);
```
Comment (sonnyp) [inline server/backend/server.js:84]: you're returning on error in the go version

File: server/backend/server.js [hunk 2]
```diff
@@ -76,33 +75,32 @@ var findRoad = (parsed, db, callback) => {
...
-
-        callback(resp);
-      });
+    if (!segment) {
+      if (err) {
+        // Collection read error
+        console.log(err);
+        resp.online = 0;
+      }
+      return callback(resp);
```
Comment (sonnyp) [inline server/backend/server.js:84]: you're returning on error in the go version

**Comment by aksiksi** [comment]
Otherwise nice change! I didn't notice that weird logic :smile:

**Comment by aksiksi** [comment]
Thanks for your suggestion, but we've merged in a [better approach](https://github.com/aksiksi/roadomatic/commit/cb2bc635ba9dbf8eedb6852320031779c17a1e47) which uses Promises instead.

**Pull request closed without merging**